kafka/src/main/java/org/apache/apex/malhar/kafka/KafkaConsumerWrapper.java

```diff
@@ -90,6 +92,19 @@
 
   private boolean waitForReplay = false;
 
+  private final List<Future<?>> kafkaConsumerThreads = new ArrayList<>();
+
+  public boolean hasAnyKafkaReaderThreadDied()
+  {
+    for (Future<?> future : kafkaConsumerThreads) {
+      if (future.isDone() || future.isCancelled()) {
+        return false;
```

Comment:
You have flipped true/false and also need to fix the check when calling this method.